specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-09-01/clusters.json

```diff
@@ -0,0 +1,942 @@
+          "readOnly": true
+        },
+        "id": {
+          "description": "Id of the node in the cluster.",
+          "type": "number",
+          "readOnly": true
+        },
+        "windowsServerSubscription": {
+          "description": "State of Windows Server Subscription.",
+          "type": "string",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "x-ms-enum": {
+            "name": "WindowsServerSubscription",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "nodeType": {
+          "description": "Type of the cluster node hardware.",
+          "type": "string",
+          "enum": [
+            "FirstParty",
+            "ThirdParty"
+          ],
+          "x-ms-enum": {
+            "name": "ClusterNodeType",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "manufacturer": {
+          "description": "Manufacturer of the cluster node hardware.",
+          "type": "string",
+          "readOnly": true
+        },
+        "model": {
+          "description": "Model name of the cluster node hardware.",
+          "type": "string",
+          "readOnly": true
+        },
+        "osName": {
+          "description": "Operating system running on the cluster node.",
+          "type": "string",
+          "readOnly": true
+        },
+        "osVersion": {
+          "description": "Version of the operating system running on the cluster node.",
+          "type": "string",
+          "readOnly": true
+        },
+        "osDisplayVersion": {
+          "description": "Display version of the operating system running on the cluster node.",
+          "type": "string",
+          "readOnly": true
+        },
+        "serialNumber": {
+          "description": "Immutable id of the cluster node.",
+          "type": "string",
+          "readOnly": true
+        },
+        "coreCount": {
+          "description": "Number of physical cores on the cluster node.",
+          "type": "number",
+          "readOnly": true
+        },
+        "memoryInGiB": {
+          "description": "Total available memory on the cluster node (in GiB).",
+          "type": "number",
+          "readOnly": true
+        }
+      }
+    },
+    "RawCertificateData": {
+      "type": "object",
+      "properties": {
+        "certificates": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "UploadCertificateRequest": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RawCertificateData"
+        }
+      },
+      "additionalProperties": false
+    },
+    "SoftwareAssuranceChangeRequestProperties": {
+      "type": "object",
+      "properties": {
+        "softwareAssuranceIntent": {
+          "type": "string"
+        }
+      },
```

Comment:
Does this accept any free form string ? Please consider if this can be modelled as enum ? 


[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

String types that should have enums
If a property has a finite set of values (enforced by the service), the string property should be declared as an enum type in the Swagger (using the x-ms-enum annotation) to enable client-side input validation and discovery.

Reply:
Sure @suhasaraos  Changed that.